File: BroadcastReceivers/CustomBroadcast/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.java
package com.tomerpacific.broadcastreceiver;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button sendBroadcastBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        sendBroadcastBtn = findViewById(R.id.sendBroadcastBtn);
        sendBroadcastBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                intent.setAction("com.example.broadcast.MY_NOTIFICATION");
                intent.putExtra("data","Hello World!");
                sendBroadcast(intent);
            }
        });
    }
}

File: BroadcastReceivers/CustomBroadcast/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.kt
package com.tomerpacific.broadcastreceiver

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import android.content.IntentFilter
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private val CUSTOM_INTENT_ACTION: String = "com.example.broadcast.MY_NOTIFICATION"

    private val receiver: MyBroadcastReceiver = MyBroadcastReceiver()
    private val intentFilter: IntentFilter = IntentFilter(CUSTOM_INTENT_ACTION)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<Button?>(R.id.sendBroadcastBtn).apply {
            setOnClickListener {
                val intent = Intent()
                intent.action = CUSTOM_INTENT_ACTION
                intent.putExtra("data", "Hello World!")
                sendBroadcast(intent)
            }
        }
    }

    override fun onResume() {
        super.onResume()
        registerReceiver(receiver, intentFilter)
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(receiver)
    }
}